app/src/main/java/com/lightricks/feedexercise/database/DataAccessObjects.kt

```diff
@@ -1,5 +1,27 @@
 package com.lightricks.feedexercise.database
 
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import io.reactivex.rxjava3.core.Completable
+import io.reactivex.rxjava3.core.Observable
+
 /***
  * todo: add Room's Data Access Object interface(s) here
  */
+
+@Dao
+interface FeedItemDao {
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun insertAll(feedItems: List<FeedItemEntity>): Completable
+
+    @Query("DELETE FROM FeedItems")
+    fun deleteAll(): Completable
+
+    @Query("SELECT * FROM FeedItems")
+    fun getAll(): Observable<List<FeedItemEntity>>
+
+    @Query("SELECT COUNT(id) FROM FeedItems")
+    fun getItemsCount(): Int
```

Comment:
I would add as a suffix `Sync` as keyword. That is because all the other methods of the class are async apart this one. So in order to highlight this difference I would add the Sync keyword. `getItemsCountSync`

Reply:
Maybe I should add the suffix Async to the other methods, since the default behavior of the language is sync methods?